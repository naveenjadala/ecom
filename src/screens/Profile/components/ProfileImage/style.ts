import {StyleSheet} from 'react-native';

export const profileImgStyles = StyleSheet.create({
  profileImage: {
    width: 100,
    height: 100,
    borderRadius: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
