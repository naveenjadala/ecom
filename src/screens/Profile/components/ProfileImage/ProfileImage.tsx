import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import {profileImgStyles} from './style';
import {Image} from 'react-native-svg';
import Avatar from '../../../../assets/images/avatar.svg';

type Props = {};

/**
 * Renders the profile image component.
 *
 * @param {Props} props - the props object containing the necessary data for rendering the component.
 * @return {JSX.Element} - the rendered profile image component.
 */
const ProfileImage = (props: Props) => {
  return (
    <TouchableOpacity
      style={profileImgStyles.profileImage}
      accessibilityRole="button">
      <Avatar width={100} height={100} />
    </TouchableOpacity>
  );
};

export default ProfileImage;

const styles = StyleSheet.create({});
