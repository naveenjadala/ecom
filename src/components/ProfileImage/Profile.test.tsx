import {render, screen} from '../../test/test-utils';
import ProfileImage from './ProfileImage';

test('should Profile snapshot', () => {
  render(
    <ProfileImage
      onClick={function (): void | undefined {
        throw new Error('Function not implemented.');
      }}
    />,
  );
  expect(screen).toMatchSnapshot();
});
describe('should render Profile Image component', () => {
  test('should render touchable Image', () => {
    render(
      <ProfileImage
        onClick={function (): void | undefined {
          throw new Error('Function not implemented.');
        }}
      />,
    );
    const getClickableImageView = screen.getByRole('button');
    expect(getClickableImageView).toBeTruthy();
  });
});
