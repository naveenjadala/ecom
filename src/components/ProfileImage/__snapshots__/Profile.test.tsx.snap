// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Profile snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "minHeight": "100%",
      "minWidth": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 100,
        "height": 100,
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
    testID="profileImage"
  >
    <test-file-stub
      height={100}
      width={100}
    />
  </View>
</RCTSafeAreaView>
`;
